zipline/finance/risk/cumulative.py

```diff
@@ -312,10 +312,18 @@ def update(self, dt, algorithm_returns, benchmark_returns, account):
         self.excess_returns[self.latest_dt] = (
             self.algorithm_cumulative_returns[self.latest_dt] -
             self.treasury_period_return)
-        self.metrics.beta[dt] = self.calculate_beta()
+        dt_index = self.metrics.index.searchsorted(dt)
+        algorithm_returns = self.algorithm_returns.values[:dt_index + 1]
```

Comment:
I believe so.

IIRC, historically the `dropna` was there both for the righthand trimming of the values, and dealing with possible holes in the data because of not being triggered every bar; however, now that the benchmark has been acting as the 'clock' for triggering this logic that secondary reason shouldn't be needed anymore.